packages/pwa-kit-react-sdk/src/ssr/universal/components/with-react-query/index.test.js

```diff
@@ -237,4 +237,255 @@ describe('withReactQuery', function () {
+            },
+            __performanceTimer: mockPerformanceTimer
+        }
+
+        const Component = withReactQuery({})
+
+        // Reset the logger mock to ensure clean state
+        logger.error.mockClear()
+
+        await Component.doInitAppState({
+            res,
+            appJSX: <div>Test</div>
+        })
+
+        // Verify that the error is logged with correct parameters
+        expect(logger.error).toHaveBeenCalledTimes(1)
+        expect(logger.error).toHaveBeenCalledWith('Query during SSR results in an error', {
+            namespace: 'with-react-query.doInitAppState',
+            additionalProperties: {
+                queryHash: 'failing-hash',
+                error: mockError
+            }
+        })
+    })
+
+    test('Multiple queries with mixed success/error status - all performance markers ended and errors logged', async () => {
+        const mockPerformanceTimer = {
+            mark: jest.fn()
+        }
+
+        const mockError = new Error('Second query failed')
+        const mockQueryCache = {
+            getAll: jest.fn().mockReturnValue([
+                {
+                    options: {enabled: true},
+                    meta: {displayName: 'SuccessQuery'},
+                    queryHash: 'success-hash',
+                    fetch: jest.fn().mockResolvedValue({data: 'success'})
+                },
+                {
+                    options: {enabled: true},
+                    meta: {displayName: 'ErrorQuery'},
+                    queryHash: 'error-hash',
+                    fetch: jest.fn().mockRejectedValue(mockError)
+                },
+                {
+                    options: {enabled: true},
+                    meta: {},
+                    queryHash: 'another-success-hash',
+                    fetch: jest.fn().mockResolvedValue({data: 'another success'})
+                }
+            ])
+        }
+
+        const mockQueryClient = {
+            getQueryCache: jest.fn().mockReturnValue(mockQueryCache)
+        }
+
+        const res = {
+            locals: {
+                __queryClient: mockQueryClient
+            },
+            __performanceTimer: mockPerformanceTimer
+        }
+
+        const Component = withReactQuery({})
+
+        // Reset the logger mock to ensure clean state
+        logger.error.mockClear()
+
+        await Component.doInitAppState({
+            res,
+            appJSX: <div>Test</div>
+        })
+
+        // Verify all timers are started
+        expect(mockPerformanceTimer.mark).toHaveBeenCalledWith(
+            `${PERFORMANCE_MARKS.reactQueryUseQuery}.SuccessQuery-0`,
+            'start'
+        )
+        expect(mockPerformanceTimer.mark).toHaveBeenCalledWith(
+            `${PERFORMANCE_MARKS.reactQueryUseQuery}.ErrorQuery-1`,
+            'start'
+        )
+        expect(mockPerformanceTimer.mark).toHaveBeenCalledWith(
+            `${PERFORMANCE_MARKS.reactQueryUseQuery}.2`,
```

Comment:
You might want to drop a note here with respect to the magic number "2". I'm assuming that because the component doesn't have a display name it's using the index. It might help make someones life easier in the future. Bonus to add a test that calls this out explicitly.

Reply:
Yup, I'll add a comment there to explain.

As for adding a test to verify with and without `displayName`, we already have such a test here: https://github.com/SalesforceCommerceCloud/pwa-kit/blob/189c41f46edcf082d9f9cf8d7e1eef983c5cecd0/packages/pwa-kit-react-sdk/src/ssr/universal/components/with-react-query/index.test.js#L153